engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/IsSameObjectNode.java

```diff
@@ -50,7 +50,13 @@ boolean isSameType(Type typeLeft, Type typeRight) {
    *
    * @return True if the qualified names of the meta objects are same.
    */
-  @Specialization(guards = {"interop.isMetaObject(metaLeft)", "interop.isMetaObject(metaRight)"})
+  @Specialization(
+      guards = {
+        "!interop.isNull(metaLeft)",
+        "interop.isMetaObject(metaLeft)",
+        "!interop.isNull(metaRight)",
+        "interop.isMetaObject(metaRight)"
+      })
```

Comment:
Do we know how to reproduce the problem?

Reply:
run stdlib `test/Tests` with assertions on. I haven't been able to reduce it to something small yet.